lib/devise/models/database_authenticatable.rb

```diff
@@ -37,11 +37,12 @@ def self.required_fields(klass)
       # the hashed password.
       def password=(new_password)
         @password = new_password
-        self.encrypted_password = password_digest(@password) if @password.present?
+        self.encrypted_password = password_digest(@password)
```

Comment:
> So, I am generating password digest for nil values as well.

We shouldn't - `nil` values should set the `encrypted_password` as `nil` otherwise the validations won't be aware that the provided value is `nil`.